docs/examples/intro_to_kernels.py

```diff
@@ -0,0 +1,369 @@
+
+# %% [markdown]
+# We can see that optimising the hyperparameters by minimising the negative log marginal likelihood of the data results in a more faithful fit of the GP to the data. In particular, we can observe that the GP using optimised hyperparameters is more accurately able to reflect uncertainty in its predictions, as opposed to the GP using the default parameters, which is overconfident in its predictions.
+#
+# The lengthscale, $\kappa$, and variance, $\sigma^2$, are shown below, both before and after optimisation:
+
+# %%
+no_opt_lengthscale = no_opt_posterior.prior.kernel.lengthscale
+no_opt_variance = no_opt_posterior.prior.kernel.variance
+opt_lengthscale = opt_posterior.prior.kernel.lengthscale
+opt_variance = opt_posterior.prior.kernel.variance
+
+print(f"Optimised Lengthscale: {opt_lengthscale} and Variance: {opt_variance}")
+print(
+    f"Non-Optimised Lengthscale: {no_opt_lengthscale} and Variance: {no_opt_variance}"
+)
+
+# %% [markdown]
+# ## Expressing Other Priors with Different Kernels
+
+# %% [markdown]
+# Whilst the Matérn kernels are often used as a first choice of kernel, and they often perform well due to their smoothing properties often being well-aligned with the properties of the underlying function being modelled, sometimes more prior knowledge is known about the function being modelled. For instance, it may be known that the function being modelled is *periodic*. In this case, a suitable kernel choice would be the *periodic* kernel:
+#
+# $$k(\mathbf{x}, \mathbf{x}') = \sigma^2 \exp \left( -\frac{1}{2} \sum_{i=1}^{D} \left(\frac{\sin (\pi (\mathbf{x}_i - \mathbf{x}_i')/p)}{\ell}\right)^2 \right)$$
+#
+# with $D$ being the dimensionality of the inputs.
+#
+# Below we show $10$ samples drawn from a GP prior using the periodic kernel:
+
+# %%
+mean = gpx.mean_functions.Zero()
+kernel = gpx.kernels.Periodic()
+prior = gpx.Prior(mean_function=mean, kernel=kernel)
+
+x = jnp.linspace(-3.0, 3.0, num=200).reshape(-1, 1)
+rv = prior(x)
+y = rv.sample(seed=key, sample_shape=(10,))
+
+fig, ax = plt.subplots()
+ax.plot(x, y.T, alpha=0.7)
+ax.set_title("Samples from the Periodic Kernel")
+plt.show()
+
+# %% [markdown]
+# In other scenarios, it may be known that the underlying function is *linear*, in which case the *linear* kernel would be a suitable choice:
+#
+# $$k(\mathbf{x}, \mathbf{x}') = \sigma^2 \mathbf{x}^\top \mathbf{x}'$$
+#
+# Unlike the kernels shown above, the linear kernel is *not* stationary, and so it is not invariant to translations in the input space.
+#
+# Below we show $10$ samples drawn from a GP prior using the linear kernel:
+
+# %%
+mean = gpx.mean_functions.Zero()
+kernel = gpx.kernels.Linear()
+prior = gpx.Prior(mean_function=mean, kernel=kernel)
+
+x = jnp.linspace(-3.0, 3.0, num=200).reshape(-1, 1)
+rv = prior(x)
+y = rv.sample(seed=key, sample_shape=(10,))
+
+fig, ax = plt.subplots()
+ax.plot(x, y.T, alpha=0.7)
+ax.set_title("Samples from the Linear Kernel")
+plt.show()
+
```

Comment:
It would be great if you added a couple more examples. Perhaps show what happens when you add or multiple kernels together?

Reply:
I think it would be super cool to end with a recreation of the "Mauna Loa atmospheric CO2" example of Section 5 of https://proceedings.mlr.press/v28/duvenaud13.pdf

This is very much "on brand" with the environmental focus of GpJax and demonstrates loads of cool stuff, i.e. its a periodic/sum of multiple base kernels? What do you think @thomaspinder

Reply:
![image](https://github.com/JaxGaussianProcesses/GPJax/assets/32096840/108b7cb1-ce42-4de7-8850-1c8849d74056)

Reply:
The airline dataset is perhaps even more interesting from a modelling perspective 
![image](https://github.com/JaxGaussianProcesses/GPJax/assets/32096840/c08c4379-ee32-4a0b-8107-1ebecaa99521)

Reply:
> It would be great if you added a couple more examples. Perhaps show what happens when you add or multiple kernels together?

I have now added a section on composing kernels through addition/multiplication of existing kernels.

Reply:
> I think it would be super cool to end with a recreation of the "Mauna Loa atmospheric CO2" example of Section 5 of https://proceedings.mlr.press/v28/duvenaud13.pdf
> 
> This is very much "on brand" with the environmental focus of GpJax and demonstrates loads of cool stuff, i.e. its a periodic/sum of multiple base kernels? What do you think @thomaspinder

Nice idea, I now have a section where we fit the CO2 dataset with a custom kernel.